src/helm/benchmark/adaptation/adapters/adapter.py

```diff
@@ -0,0 +1,74 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+import numpy as np
+
+from helm.benchmark.adaptation.adapter_spec import AdapterSpec
+from helm.benchmark.adaptation.request_state import RequestState
+from helm.benchmark.adaptation.scenario_state import ScenarioState
+from helm.benchmark.scenarios.scenario import Instance, TRAIN_SPLIT, EVAL_SPLITS
+from helm.benchmark.window_services.tokenizer_service import TokenizerService
+from helm.benchmark.window_services.window_service import WindowService
+from helm.benchmark.window_services.window_service_factory import WindowServiceFactory
+from helm.common.hierarchical_logger import hlog
+
+
+class Adapter(ABC):
+    """
+    An `Adapter`, guided by the `AdapterSpec`, takes a `Scenario` and produces
+    a `ScenarioState`. This is where prompt hacking logic should live.
+    """
+
+    def __init__(self, adapter_spec: AdapterSpec, tokenizer_service: TokenizerService):
+        self.adapter_spec: AdapterSpec = adapter_spec
+        self.window_service: WindowService = WindowServiceFactory.get_window_service(
+            adapter_spec.model, tokenizer_service
+        )
+
+    @abstractmethod
+    def adapt(self, instances: List[Instance], parallelism: int) -> ScenarioState:
+        """
+        Takes a a list of `Instance`s and returns a `ScenarioState` with the
+        list of corresponding `RequestState`s.
+        """
+        pass
+
+    @abstractmethod
+    def generate_requests(self, eval_instance: Instance) -> List[RequestState]:
+        """
+        Given a validation or test `Instance`, generates one or more `RequestState`s.
+        """
+        pass
+
+    def sample_instances(self, instances: List[Instance]) -> List[Instance]:
```

Comment:
This seems like a strange abstraction especially given the function name - I would expect sampling instances to sample instances, not to keep train examples around.

Reply:
I renamed it `get_run_instances` because we're getting the instances needed for this run based on the `AdapterSpec`. I also clarified in the method comments.